src/Recommendations/Process/Hydration.php

```diff
@@ -73,18 +74,33 @@ public function getPodcastEpisodeChapterById($id): PodcastEpisodeChapterModel
         return new PodcastEpisodeChapterModel($episode, $chapter);
     }
 
+    public function getExternalArticleByOriginalArticleId($id): ExternalArticle
+    {
+        list($originalArticleId, $relatedIndex) = explode('-', $id);
+        $relatedIndex = (int) $relatedIndex;
+
+/** @var ExternalArticle $episode */
+        // TODO: this uses sdk but it should really go through SingleItemRepository (doesn't have a method for this) or in any case through a cache
+        $externalArticle = $this->sdk->getRelatedArticles($originalArticleId)[$relatedIndex];
```

Comment:
Will this cause issues if a related article is removed from the API? Not sure if this should be more resilient to errors, what do you think?